### PR TITLE
Add build bots for ml-driven optimization policies.

### PR DESCRIPTION
**Pull request by mtrofin**

### CODE REVIEW

**Comment by mtrofin** [comment]
What is the test running procedure?

File: buildbot/osuosl/master/config/builders.py [hunk 1]
```diff
@@ -1646,6 +1646,43 @@ def _get_documentation_builders():
              }
            ]
 
+# Builders for ML-driven compiler optimizations.
+def _get_ml_compiler_opt_builders():
+    common_extra_args = ["-DCMAKE_BUILD_TYPE=Release", ]
+    return [
+        {'name': "ml-opt-dev-x86-64",
+         'mergeRequests': False,
+         'slavenames':["ml-opt-dev-x86-64-b1"],
```
Comment (davidxl) [inline buildbot/osuosl/master/config/builders.py:1655]: what is the b1 suffix about?

File: buildbot/osuosl/master/config/builders.py [hunk 2]
```diff
@@ -1646,6 +1646,43 @@ def _get_documentation_builders():
              }
            ]
 
+# Builders for ML-driven compiler optimizations.
+def _get_ml_compiler_opt_builders():
+    common_extra_args = ["-DCMAKE_BUILD_TYPE=Release", ]
+    return [
+        {'name': "ml-opt-dev-x86-64",
```
Comment (davidxl) [inline buildbot/osuosl/master/config/builders.py:1653]: add a comment about this config?

File: buildbot/osuosl/master/config/builders.py [hunk 3]
```diff
@@ -1646,6 +1646,43 @@ def _get_documentation_builders():
...
+         'mergeRequests': False,
+         'slavenames':["ml-opt-dev-x86-64-b1"],
+         'builddir':"ml-opt-dev-x86-64-b1",
+         'factory': UnifiedTreeBuilder.getCmakeWithNinjaBuildFactory(
+                        clean=True,
+                        depends_on_projects=['llvm'],
+                        extra_configure_args=common_extra_args + [
+                            "-DTENSORFLOW_API_PATH=${TENSORFLOW_API_PATH}"
+                        ])},
+        {'name': "ml-opt-rel-x86-64",
```
Comment (davidxl) [inline buildbot/osuosl/master/config/builders.py:1663]: add a comment.

File: buildbot/osuosl/master/config/builders.py [hunk 4]
```diff
@@ -1646,6 +1646,43 @@ def _get_documentation_builders():
...
+         'mergeRequests': False,
+         'slavenames':["ml-opt-rel-x86-64-b1"],
+         'builddir':"ml-opt-rel-x86-64-b1",
+         'factory': UnifiedTreeBuilder.getCmakeWithNinjaBuildFactory(
+                        clean=True,
+                        depends_on_projects=['llvm'],
+                        extra_configure_args=common_extra_args + [
+                            "-DTENSORFLOW_AOT_PATH=${TENSORFLOW_AOT_PATH}"
+                        ])},
+        {'name': "ml-opt-devrel-x86-64",
```
Comment (davidxl) [inline buildbot/osuosl/master/config/builders.py:1673]: comment here.

File: buildbot/osuosl/master/config/builders.py (hunk 1 above)
Comment (mtrofin) [reply buildbot/osuosl/master/config/builders.py:1655]: if we later need more bots - it was based on what I saw sanitizers did.

File: buildbot/osuosl/master/config/builders.py (hunk 2 above)
Comment (mtrofin) [reply buildbot/osuosl/master/config/builders.py:1653]: Done

File: buildbot/osuosl/master/config/builders.py (hunk 3 above)
Comment (mtrofin) [reply buildbot/osuosl/master/config/builders.py:1663]: Done

File: buildbot/osuosl/master/config/builders.py (hunk 4 above)
Comment (mtrofin) [reply buildbot/osuosl/master/config/builders.py:1673]: Done

**Comment by mtrofin** [comment]
Closing - see https://reviews.llvm.org/D81044 instead.

**Pull request closed without merging**